### ISSUE TITLE
Названия и расположения директорий сниппетов

### ISSUE THREAD

**Issue by lord-alfred** [ZennoHelpers/Snippets#1]
### UPDATE:
**Актуальный [Список директорий](https://github.com/ZennoHelpers/Snippets/wiki/%D0%A1%D0%BF%D0%B8%D1%81%D0%BE%D0%BA-%D0%B4%D0%B8%D1%80%D0%B5%D0%BA%D1%82%D0%BE%D1%80%D0%B8%D0%B9) находится в wiki.**

---

Предлагаю следующую _начальную_ структуру директорий (коллекций) для сниппетов, чтобы сразу же немного стандартизировать их расположение для всех желающих добавить свой сниппет.

- [Анонимность]
- [Базы данных]
- [Браузер]
- [Капчи]
- [Картинки]
- [Логика]
- [Почта]
- [Прокси]
- [Регулярки]
- [Списки]
- [Таблицы]
- [Эмуляция]
- [FTP]
- [HTTP]
- [XPath]

В каждой коллекции могут быть подколлекции, но опять же - лучше заранее обсудить их названия (чтоб не было путаницы в дальнейшем).

**Каких коллекций верхнего уровня ещё не хватает?**

Предлагаю обсудить этот issue до понедельника (16.09.2019) и если будет однозначное решение/согласие, то создать недостающие пустые директории к тому времени в репозитории.

**Comment by ghost** [comment]
[Запросы] - туда войдут [FTP], [HTTP], [WebSocket]
[Почта]
[Профиль]
[Директории, Файлы]
[Лог, Оповещения]
[Ввод] - [Мышь], [Набор]
[Потоки] - управление и связанное
[Regex] - какие-нибудь фичи
[Текст]
[JS] - куда без него тут)

[xPath] - войдёт в [Браузер] и [Текст], смотря где используется

У меня есть ещё подпапка [ZP] для директории со сниппетами, не относящимися к зенке, а вообще к C#. Выше уже разделение на C#, F#, JS, Regex. Но так как у нас зенка, а основной C#, то остальное будет либо в подпапках, либо с остальными сниппетами.

**Comment by lord-alfred** [comment]
[Ввод] заменит коллекцию [Эмуляция], название которой мне не нравится (слишком широкий смысл).

По поводу [XPath], который входит в коллекции [Браузер] и [Текст] - я вот не могу сходу сказать кейс, когда необходимо разделение сниппетов от XPath на браузер/безбраузер (если только мы не говорим сразу же об использовании его в других либах). Просто не совсем понимаю зачем в коллекции [Текст] он нужен?

**Comment by ghost** [comment]
xPath теперь можно использовать не только в браузере, а например, в [ParseByXpath(String,String,String)](https://help.zennolab.com/en/v5/zennoposter/5.30.0.0/webframe.html#topic832.html) там работа чисто со строкой и на выходе коллекция строк.

**Comment by ghost** [comment]
[Анонимность] вопросы создаёт. Лучше описать её в wiki со ссылками на файлы.

**Comment by ghost** [comment]
Надо решить над расположением [Капчи] и [Картинки]. Методы рассчитанные на работу в браузере, логичнее видеть в [Браузер]/[Капчи], [Браузер]/[Картинки]. Связанное с локальной обработкой (как файлы), можно пока оставить в корне.

Ещё есть момент, что назначение сниппетов в [Капчи] и [Картинки] могут совпадать.

**Comment by lord-alfred** [comment]
Мне кажется, что как раз будет путаница если начать разделять "Браузер/Капчи" и "Запросы/Капчи", т.к. есть контр-примеры:
- отправка base64 капчи на распознавание в capmonster 
- изменения размера изображений

Оба примера могут быть как с использованием браузера, так и без него. Поэтому, думаю, что нужно как раз глобальные коллекции для каптч и картинок. Возможно, тоже самое стоит сделать с XPath, но лучше ты насчет него скажи, т.к. я не работал с вышеприведенным методом.

**Comment by leotop** [comment]
Работа с файловой системой, удалить, переместить, список, обход, куда?
[Файловая система] ?

**Comment by lord-alfred** [comment]
Работа с файловой системой будет в коллекции: [Директории, Файлы] - @Zymlex уже отправил PR, где есть эта коллекция.

По поводу "список, обход, куда" - не понял )

**Comment by leotop** [comment]
список всех файлов, список всех файлов с расширением
рекурсивный обход

ок, ожидаю появления разделов

**Comment by lord-alfred** [comment]
Да, это будет в коллекции с именем папки "[Директории, Файлы]".

Ждать не обязательно, уже сейчас можешь оформить PR - там не должно быть конфликтов при мерже.

**Comment by lord-alfred** [comment]
### UPDATE:
**Актуальный [Список директорий](https://github.com/ZennoHelpers/Snippets/wiki/%D0%A1%D0%BF%D0%B8%D1%81%D0%BE%D0%BA-%D0%B4%D0%B8%D1%80%D0%B5%D0%BA%D1%82%D0%BE%D1%80%D0%B8%D0%B9) находится в wiki.**

---

### По итогам обсуждения стандартная структура директорий следующая:

- [Анонимность]
- [Базы данных]
- [Браузер]
- [Ввод] - (туда войдут [Мышь], [Набор])
- [Даты, Время]
- [Директории, Файлы]
- [Запросы] (туда войдут [FTP], [HTTP], [WebSocket])
- [Капчи]
- [Картинки]
- [Лог, Оповещения]
- [Логика]
- [Потоки]
- [Почта]
- [Прокси]
- [Профиль]
- [Списки]
- [Сторонний GUI]
- [Таблицы]
- [Текст]
- [JavaScript]
- [Regex]
- [XPath]

Если будет необходимо добавить какие-то другие коллекции **верхнего** уровня, то обсудить их названия можно в этом issue.

А пока что давайте располагать сниппеты согласно данной структуре, чтобы всё было минимально стандартизировано изначально. В дальнейшем ещё 10 раз можем всё поменять, главное сейчас между собой не путаться :)

_PS: вначале хотел создать пустые директории в репозитории, которых не хватает, но сейчас понимаю что это будет лишним. Ссылка из readme есть сюда, новички увидят и расположат свои сниппеты как надо, если нет - мы поправим в code review._

**Comment by AZANIR** [comment]
Яб добавил [время] [даты] итд... обработка , часто нужно для сравнения и не только ..

**Comment by lord-alfred** [comment]
Хорошая мысль! Для обработки дат и времени, думаю, логично использовать одну коллекцию: [Даты, время]

**Comment by AZANIR** [comment]
Также по работе с БД у нас нет тут ничего

**Comment by lord-alfred** [comment]
Как это? Есть коллекция "[Базы данных]"

**Comment by AZANIR** [comment]
упс не увидел )

**Comment by lord-alfred** [comment]
Решил, что удобнее будет если список коллекций будет лежать в wiki: [Список директорий](https://github.com/ZennoHelpers/Snippets/wiki/%D0%A1%D0%BF%D0%B8%D1%81%D0%BE%D0%BA-%D0%B4%D0%B8%D1%80%D0%B5%D0%BA%D1%82%D0%BE%D1%80%D0%B8%D0%B9).

**Здесь в дальнейшем ожидаю, что будут запросы на добавление новых коллекций или обсуждение необходимости каких-то из тех, что уже есть.**

**Comment by MeteorBurn** [comment]
Я бы сменил категорию "Картинки" на "Изображения" или "Работа с изображением".

**Comment by lord-alfred** [comment]
Сменить на "Изображения" - имеет смысл, а вот добавлять префикс "Работа с" - вряд ли (везде какая-то "работа" :) ).
Что думают другие участники?

**Comment by MeteorBurn** [comment]
> Сменить на "Изображения" - имеет смысл, а вот добавлять префикс "Работа с" - вряд ли (везде какая-то "работа" :) ).
> Что думают другие участники?

без разницы, "работа" или нет.
Картинки это в телефоне.

**Comment by ghost** [comment]
> Я бы сменил категорию "Картинки" на "Изображения"

Можно и так.
> Картинки это в телефоне.

Картинки ещё в браузере) Кстати, папка там отдельно пока.

**Comment by DmitryAkDev** [comment]
Нужен раздел Окружение (Enviroment) залил пару сниппетов, которые туда идеально заходят, а в ином случае лежали бы в корне. Туда же стоит забрасывать снипы дергающие все "внешние" данные из системы, но не относящиеся к конкретным подразделам.

**Comment by lord-alfred** [comment]
Добавил коллекцию [Окружение].

По [Изображениям] - думаю, логично что в корне они именно так и должны называться, а в браузере пусть остаются [Картинки]. Но нужен PR от кого-то с переименованием коллекции [Картинки] в корне в [Изображения]

**Comment by ghost** [comment]
Не будем больше называть директории и папки - коллекциями.
Раздел, категория, но не коллекция :)
